src/arrayMethodJoin.js

```diff
@@ -5,8 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
-  };
+    let string = "";
+    let division = separator;
+    for (let i = 0; i < this.length; i++) {
+      if (division === undefined) {
+        division = ',';
+      }
+      if (division === null) {
+        division = 'null';
+      }
+      if (this[i] !== undefined && this[i] !== null) {
+        string += this[i];
+      }
+      if (this[i] !== this.length -1 && this.length > 1 && this[i] !== '') {
+        string += division;
+      }
```

Comment:
do you need all these conditions here?
we may check only if current element is not the last one

Reply:
if delete conditions ( this.length > 1 && this[i] !== '') some tests fail.